x/evm/keeper/grpc_query.go

```diff
@@ -577,3 +594,11 @@ func (k Keeper) BaseFee(c context.Context, _ *types.QueryBaseFeeRequest) (*types
 
 	return res, nil
 }
+
+// GetChainID parse chainID from current context if not provided
+func GetChainID(ctx sdk.Context, chainID *big.Int) (*big.Int, error) {
```

Comment:
Since this implementation prefers supplied `chainID` over context's one so it can't be used when `k.ApplyMessageWithConfig` with `commit = true` or smt like that due to chain fork replay attack. Please add comment to this method, to clarify if my opinion is correct

Reply:
imo, implement `chain id` as a fallback value is safer.

I mean when, somehow the chain id in context is empty or zero, use the supplied chain id, instead of priority the supplied value and open a tweak for someone control the flow

Reply:
> supplied chainID over context's one so it can't be used when k.ApplyMessageWithConfig with commit = true or smt like that due to chain fork replay attack

depending on where does the supplied one come from, currently it's only used in grpc query, so it has nothing to do with consensus state machine, we can make it private if worry about misuse.

Reply:
For grpc query, the information passed from json-rpc is parsed from correct block header, while the `ctx` is from arbitrary block, so the provided one is more legitimate. i've changed it to private to avoid using in consensus state machine.